.github/workflows/ci.yaml

```diff
@@ -436,6 +439,112 @@ jobs:
           name: e2e-screenshots
           path: test/e2e/cypress/screenshots/
 
+  check-regression-label:
+    name: Check if the environment creation criteria are met, store in the job output
+    runs-on: ubuntu-latest
+    outputs:
+      run_regression_test: ${{ steps.check.outputs.run_regression_test }}
+    steps:
+      - id: check
+        run: echo "run_regression_test=${{ contains(fromJson(env.MANTAINERS), github.event.sender.login) && contains(github.event.pull_request.labels.*.name, env.RG_TEST_LABEL) }}" >> "$GITHUB_OUTPUT"
+
+  regression-test-e2e:
+    name: Regression tests
+    needs: [elixir-deps, npm-deps, npm-e2e-deps, api-bc-check]
+    runs-on: ubuntu-22.04
+    env:
+      MIX_ENV: dev
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.12.1
+        with:
+          access_token: ${{ github.token }}
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Setup
+        uses: erlef/setup-beam@v1
+        with:
+          elixir-version: ${{ env.ELIXIR_VERSION }}
+          otp-version: ${{ env.OTP_VERSION }}
+        env:
+          ImageOS: ubuntu20
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+      - name: Retrieve Cached Dependencies
+        uses: actions/cache@v4
+        id: mix-cache
+        with:
+          path: |
+            deps
+            _build/dev
+            priv/plts
+          key: ${{ runner.os }}-${{ env.OTP_VERSION }}-${{ env.ELIXIR_VERSION }}-${{ hashFiles('mix.lock') }}
+      - name: Retrieve NPM Cached Dependencies
+        uses: actions/cache@v4
+        id: npm-cache
+        with:
+          path: |
+            assets/node_modules
+          key: ${{ runner.os }}-${{ env.NODE_VERSION }}-${{ hashFiles('assets/package-lock.json') }}
+      - name: Retrieve E2E NPM Cached Dependencies
+        uses: actions/cache@v4
+        id: npm-e2e-cache
+        with:
+          path: |
+            test/e2e/node_modules
+          key: ${{ runner.os }}-${{ env.NODE_VERSION }}-${{ hashFiles('test/e2e/package-lock.json') }}
+      - name: Check Eslint and JS Code Format
+        run: cd test/e2e && npm run lint && npm run format:check
+      - name: "Docker compose dependencies"
+        uses: isbang/compose-action@v1.5.1
+        with:
+          compose-file: "./docker-compose.yaml"
+          down-flags: "--volumes"
+          compose-flags: "-f docker-compose.regression.yaml"
+      - name: Mix setup
+        run: mix setup
```

Comment:
`mix setup`, are you sure of this?
In all the tests I did this was not necessary, in fact, it created a lot of issues because it tries to modify the database with entries that are already modified

Reply:
I'm not sure but I got in troubles running the restores without it. I didn't look much into it as I thought it was needed, I'll double check :+1:

Reply:
The `mix setup` was there as a way of avoiding the creation of the database (mix setup would do this for us), but I've replaced it with a saner approach: when creating the `.sql` files with `pg_dump`, I used `-C` option which results in the `.sql` files also including `CREATE DATABASE` statements. This allows that during restoration, the databases are not required to be there in the first place (and allows to remove this `mix setup`)